test/common/http/filter/fault_filter_test.cc

```diff
@@ -200,6 +200,8 @@ TEST_F(FaultFilterTest, AbortWithHttpStatus) {
 
   EXPECT_CALL(runtime_.snapshot_, getInteger("fault.http.delay.fixed_duration_ms", _)).Times(0);
   EXPECT_CALL(filter_callbacks_, continueDecoding()).Times(0);
+  EXPECT_CALL(filter_callbacks_.request_info_,
```

Comment:
can you also add a test where DI injected followed by FI and make sure that both ResponseFlags are set?